src/features/auth/SignupPage.tsx

```diff
@@ -0,0 +1,270 @@
+import { useState } from "react";
+import {
+  Card,
+  CardContent,
+  TextField,
+  Button,
+  Typography,
+  Box,
+  Alert,
+  FormControlLabel,
+  Checkbox,
+  Link,
+} from "@mui/material";
+import { Link as Router } from "react-router";
+import "./SignupPage.css";
+
+const SignupPage = () => {
+  // form states:
+  const [formData, setFormData] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    phone: "",
+    dob: "",
+    password: "",
+    confirmPassword: "",
+  });
+
+  const [agreeToTerms, setAgreeToTerms] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [verificationSent, setVerificationSent] = useState(false);
+
+  //  input changes
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    const { name, value } = e.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  // Form validation
+  const validateForm = () => {
+    // Clear any previous errors
+    setError(null);
+
+    // Check if passwords match
+    if (formData.password !== formData.confirmPassword) {
+      setError("Passwords don't match");
+      return false;
+    }
+
+    // Check pass length
+    if (formData.password.length < 8) {
+      setError("Password must be at least 8 characters");
+      return false;
+    }
+
+    // Check terms agreement
+    if (!agreeToTerms) {
+      setError("You must agree to the terms and conditions");
+      return false;
+    }
+
+    return true;
+  };
+
+  // Handle form sub
+  const handleSignup = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Validate the form
+    if (!validateForm()) return;
+
+    setLoading(true);
+
+    try {
+      // Send signup data to the backend endpoint
+      const response = await fetch("/api/auth/register", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          email: formData.email,
+          password: formData.password,
+        }),
+      });
+
+      const data = await response.json();
+
+      // Handle error response
+      if (!response.ok) {
+        throw new Error(data.message || "Error creating account");
+      }
```

Comment:
suggestion: here we are catching the error but not checking the response before parsing JSON

Suggested fix:
```
if (!response.ok) {
    const errorData = await response.json().catch(() => null);
    throw new Error(errorData?.message || "Error creating account");
  }
```